### PR TITLE
General correction of language, style, and wrong links

### PR DESCRIPTION
**Pull request by safwansamsudeen**
@ErikSchierboom, I've corrected many stylistic and linguistic errors in the Building directory of the docs. This is in relation to the [Errors in Exercism docs issue](https://forum.exercism.org/t/errors-in-exercism-docs/4634) on the Forum. 

There is much more improvement left to be done both in this directory and outside of it, and I'll continue to work on this and open more pull requests: please do let me know if you want multiple commits in one PR or separate PRs. I'll ask on issues I'm doubtful on in the Forum question, not here.

Thanks for your time!

### CODE REVIEW

File: building/github/contributors-pull-request-guide.md [hunk 1]
```diff
@@ -1,41 +1,38 @@
...
 
 This documents contains some additional, Exercism-specific guidelines for different types of pull requests.
 
 ## Exercise Pull Requests
 
-Creating a pull request for a Concept Exercise or Practice Exercise can be daunting with the many rules around these types of exercises.
-For this reason, it can take a maintainer 2 to 3 hours to review and can result in dozens of comments.
-
-To help you, there'll be one primary reviewer commenting on your pull request.
+Creating a pull request for a Concept Exercise or Practice Exercise can be daunting with the many rules around these types of exercises. For this reason, it can take a maintainer two to three hours to review and can result in dozens of comments. To help you, there'll be one primary reviewer commenting on your pull request.
```
Comment (ErikSchierboom) [inline building/github/contributors-pull-request-guide.md]: Our style guide uses one line per sentence. Could you revert?

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 node_modules/
+.DS_Store
```
Comment (ErikSchierboom) [inline .gitignore]: This looks like a change that doesn't necessarily belong to this PR. Could you revert?

File: building/github/contributors-pull-request-guide.md [hunk 3]
```diff
@@ -1,41 +1,38 @@
...
 
 Don't be discouraged by the number of review comments; it is perfectly normal for an exercise to go through several rewrites before arriving at something both you and the primary reviewer are happy with.
 
 Here are some recommendations to help streamline the process even more.
 
-### Recommendation: read the documentation
+### Read the documentation
```
Comment (ErikSchierboom) [inline building/github/contributors-pull-request-guide.md:20]: I understand why you'd want to remove the prefix, but I'd have to give it a bit more thought. Could you revert for now? (same applies to the other prefix removals)

**Comment by ErikSchierboom** [comment]
As a side note: the PR title has an "and" in it, which implies that it is doing multiple things. For future PRs I would recommend opening several smaller PRs that each do one thing (one for fixing links, one for grammar, etc.)

File: building/github/gha-best-practices.md [hunk 4]
```diff
@@ -41,16 +41,16 @@ jobs:
 
 ### Consider if (third-party) actions are really needed
 
-Actions should be treated like dependencies in your favourite programming language[^1], they are code written by third party authors outside of the control of Exercism.
+Actions should be treated like dependencies in your favourite programming language <sup>1</sup>, they are code written by third party authors outside of the control of Exercism.
```
Comment (ee7) [inline building/github/gha-best-practices.md]: This was using the GitHub Flavored Markdown [footnote syntax](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#footnotes), which the website is [supposed to support](https://github.com/exercism/docs/issues/361#issuecomment-1173456332).

However, it does look like the footnote linking isn't fully working on [the corresponding page](https://exercism.org/docs/building/github/gha-best-practices). Any ideas, @ErikSchierboom?

I think we should revert this change. If it turns out that footnotes won't be fully supported in the near future, a follow-up PR can change all the footnotes.

File: building/github/gha-best-practices.md [hunk 5]
```diff
@@ -41,16 +41,16 @@ jobs:
...
 
 Therefore, you should carefully consider if introducing a new action is really worth it or if it's better to move the code into a (new) action under Exercism's control.
 
 Also consider if the action is actively maintained, e.g. by checking recent repo activity or ensuring that the action is part of an organisation rather than an individual account.
 
-Actions published by [GitHub](https://github.com/actions/) or the Exercism org can generally be considered as safe(ish) to include without special consideration.
+Actions published by [GitHub](https://github.com/actions/) or the Exercism organization can generally be considered as safe(ish) to include without special consideration.
 
-[^1]: unless the language uses the npm ecosystem.
+<sup>1</sup>: unless the language uses the `npm` ecosystem.
```
Comment (ee7) [inline building/github/gha-best-practices.md]: Same.

I'm also not convinced by the change to npm. Maybe it's best written as `npm` only when we're referring to the command line tool.

It is often written without monospacing on e.g.:

- https://www.npmjs.com/
- https://en.wikipedia.org/wiki/Npm_(software)

File: building/configlet/lint.md [hunk 6]
```diff
@@ -526,24 +526,24 @@ The `config.json` file should have the following checks:
...
+   - Lowercase prepositions, regardless of length, except when they are stressed, are used adverbially or adjectivally, or are used as conjunctions.
+   - Lowercase the words _to_ and _as_.
+   - Lowercase the second part of Latin species names.
+4. **Sentence-case string**: a non-blank string that follows the below guidelines (see https://en.wikipedia.org/wiki/Letter_case#Sentence_case):
+   - Capitalize the first word of the sentence, as well as proper nouns and other words as required by a more specific rule.
+5. **Valid `files` pattern**: A non-blank string¹ that specifies a location of a file used in an exercise, relative to the exercise's directory. A pattern may use one of the following placeholders:
    - `%{kebab_slug}`: the `kebab-case` exercise slug (e.g. `bit-manipulation`)
    - `%{snake_slug}`: the `snake_case` exercise slug (e.g. `bit_manipulation`)
    - `%{camel_slug}`: the `camelCase` exercise slug (e.g. `bitManipulation`)
    - `%{pascal_slug}`: the `PascalCase` exercise slug (e.g. `BitManipulation`)
```
Comment (ee7) [inline building/configlet/lint.md:546]: Aside: these are unordered list items, but are rendered as ordered list items on [the corresponding page on the website](https://exercism.org/docs/building/configlet/lint). I believe that they shouldn't be.

Any ideas @ErikSchierboom? Maybe we need to update some markdown rendering dependency? It could also be due to not adding a newline before the list items, which `markdownlint` complains about.

File: building/configlet/lint.md [hunk 7]
```diff
@@ -526,24 +526,24 @@ The `config.json` file should have the following checks:
...
-   > - Lowercase prepositions, regardless of length, except when they are stressed, are used adverbially or adjectivally, or are used as conjunctions.
-   > - Lowercase the words _to_ and _as_.
-   > - Lowercase the second part of Latin species names.
-4. Sentence Case string: a non-blank string that follows the below guidelines (see https://en.wikipedia.org/wiki/Letter_case#Sentence_case):
-   > - Capitalize the first word of the sentence, as well as proper nouns and other words as required by a more specific rule.
-5. Valid `"files"` pattern: A non-blank string¹ that specifies a location of a file used in an exercise, relative to the exercise's directory. A pattern may use one of the following placeholders:
+1. **Non-blank string**: a string that contains at least one non-whitespace character.
+2. **Kebab-case string**: a string that contains only characters in the range `[a-z0-9]`, optionally separated by dashes (e.g. "two-fer"). It must match the regular expression: `^[a-z0-9]+(-[a-z0-9]+)*$`
+3. **Title-case string**: a non-blank string that follows the below guidelines (from the [Chicago Manual of Style](https://en.wikipedia.org/wiki/Title_case)):
```
Comment (ee7) [inline building/configlet/lint.md]: The link text doesn't match the destination. If I click on a link that says "Chicago Manual of Style", I don't expect to be taken to a Wikipedia page on the title case, even if that page includes the Chicago Manual of Style's definiton of title case.

Can you rephrase?

Comment (ee7) [inline building/configlet/lint.md]: The appearance of "kebab-case" and "Title Case" was deliberate. Now "kebab-case" begins with a capital letter, and "Title-case" has a dash in it. I think that's worse, and it no longer matches how it's written in the earlier part of the document.

I like the change to add emphasis, but please revert the changes to casing and dash usage.

File: building/github/gha-best-practices.md (hunk 4 above)
Comment (ErikSchierboom) [reply building/github/gha-best-practices.md]: No idea why, weird!

File: building/configlet/lint.md (hunk 6 above)
Comment (ErikSchierboom) [reply building/configlet/lint.md:546]: It's almost certainly the latter

Comment (safwansamsudeen) [reply building/configlet/lint.md:546]: Huh? This is really weird. Anyway, I've added a line before the list items.

File: building/github/contributors-pull-request-guide.md (hunk 1 above)
Comment (safwansamsudeen) [reply building/github/contributors-pull-request-guide.md]: Do you mean one sentence per line? Sure!

File: building/github/contributors-pull-request-guide.md (hunk 3 above)
Comment (safwansamsudeen) [reply building/github/contributors-pull-request-guide.md:20]: I'll open a new issue on the forum for this, then.

**Comment by safwansamsudeen** [comment]
> As a side note: the PR title has an "and" in it, which implies that it is doing multiple things. For future PRs I would recommend opening several smaller PRs that each do one thing (one for fixing links, one for grammar, etc.)

Sure, Eric, I'll keep that in mind next time!

File: building/github/gha-best-practices.md (hunk 4 above)
Comment (safwansamsudeen) [reply building/github/gha-best-practices.md]: @ee7, I'll revert the change here, but shall I open a separate PR to change all footnotes to `sup`? By the way, the ^1 didn't work in VS Code either, which is why I changed to using `sup`.

File: building/github/gha-best-practices.md (hunk 5 above)
Comment (safwansamsudeen) [reply building/github/gha-best-practices.md]: That's a good point, I'm reverting it.

File: building/configlet/lint.md (hunk 7 above)
Comment (safwansamsudeen) [reply building/configlet/lint.md]: I've made it slightly clearer, and given a link that highlights that section. Is that OK?

Comment (safwansamsudeen) [reply building/configlet/lint.md]: Ah, I didn't realize the significance, sorry!

**Comment by ErikSchierboom** [comment]
Looks good. @ee7 what do you think?

File: .gitignore [hunk 8]
```diff
@@ -1 +1 @@
-node_modules/
+node_modules/
```
Comment (ee7) [inline .gitignore]: Every non-empty file should have exactly 1 final newline character. This PR shouldn't remove the final newline here.

File: building/github/contributors-pull-request-guide.md [hunk 9]
```diff
@@ -1,41 +1,39 @@
...
+Before creating a pull request, please read the [how to make a great pull request][how-to-make-a-great-pr] guide which aims to help you with the non-technical side of pull requests so that you have a great experience and help our maintaining team at the same time.
 
 This documents contains some additional, Exercism-specific guidelines for different types of pull requests.
 
 ## Exercise Pull Requests
 
-Creating a pull request for a Concept Exercise or Practice Exercise can be daunting with the many rules around these types of exercises.
-For this reason, it can take a maintainer 2 to 3 hours to review and can result in dozens of comments.
-
+Creating a pull request for a Concept Exercise or Practice Exercise can be daunting with the many rules around these types of exercises. For this reason, it can take a maintainer two to three hours to review and can result in dozens of comments.
```
Comment (ee7) [inline building/github/contributors-pull-request-guide.md]:
```suggestion
Creating a pull request for a Concept Exercise or Practice Exercise can be daunting with the many rules around these types of exercises.
For this reason, it can take a maintainer two to three hours to review and can result in dozens of comments. 
```

Let's keep "one sentence per line" where it existed already. We should have "one sentence per line" in all markdown files, but that should be addressed in a separate PR.

File: building/configlet/lint.md [hunk 10]
```diff
@@ -526,24 +526,26 @@ The `config.json` file should have the following checks:
...
-   > - Lowercase the words _to_ and _as_.
-   > - Lowercase the second part of Latin species names.
-4. Sentence Case string: a non-blank string that follows the below guidelines (see https://en.wikipedia.org/wiki/Letter_case#Sentence_case):
-   > - Capitalize the first word of the sentence, as well as proper nouns and other words as required by a more specific rule.
-5. Valid `"files"` pattern: A non-blank string¹ that specifies a location of a file used in an exercise, relative to the exercise's directory. A pattern may use one of the following placeholders:
+1. **Non-blank string**: a string that contains at least one non-whitespace character.
+2. **kebab-case string**: a string that contains only characters in the range `[a-z0-9]`, optionally separated by dashes (e.g. "two-fer"). It must match the regular expression: `^[a-z0-9]+(-[a-z0-9]+)*$`
+3. **Title Case string**: a non-blank string that follows the below guidelines (from the [Chicago Manual of Style](https://en.wikipedia.org/wiki/Title_case#Chicago%20Manual%20of%20Style:~:text=Guitar\).%5B3%5D-,Chicago%20Manual%20of%20Style,-%5Bedit%5D) of title case):
```
Comment (ee7) [inline building/configlet/lint.md]:
```suggestion
3. **Title Case string**: a non-blank string that follows the below [guidelines from the Chicago Manual of Style](https://en.wikipedia.org/wiki/Title_case#Chicago_Manual_of_Style): 
```

Improve link text and destination.

**Review by ee7 — COMMENTED** [review]
LGTM after these small changes. Sorry for the slow review.

File: building/tooling/analyzers/feedback-guidelines.md [hunk 11]
```diff
@@ -4,8 +4,8 @@ _NOTE: This spec is currently being updated._
 
 The goal of a language track on Exercism is to give people a way to achieve a
 high level of [fluency](/docs/using/product/fluency) at a low level of proficiency. We're aiming for fluency
-in the syntax, idioms, and the standard library of the language. You can read
-more about the [goal of exercism here](https://github.com/exercism/docs/blob/master/about/goal-of-exercism.md).
+in the syntax, idioms, and the standard library of the language. Read
+more about the [goal of Exercism here](https://github.com/exercism/docs/blob/master/about/goal-of-exercism.md).
```
Comment (thinkverse) [inline building/tooling/analyzers/feedback-guidelines.md:8]: This link currently doesn't lead anywhere, see #424. Linking to *getting started* could be a potential fix, or revert these current changes so as to not cause a merge conflict and let the latter PR resolve it.

```suggestion
in the syntax, idioms, and the standard library of the language. Read
more about the [goal of Exercism here](/docs/using/getting-started).
```

Comment (safwansamsudeen) [reply building/tooling/analyzers/feedback-guidelines.md:8]: Are you sure the getting started page is a valid link for "goal of Exercism"? Also, which changes should I revert to ensure that there isn't a merge conflict?

Comment (thinkverse) [reply building/tooling/analyzers/feedback-guidelines.md:8]: Not sure if it's "valid" but it's the only page I found that mentions "goal of Exercism" now. 🤷‍♂️ Maybe the team knows a better URL to point to.

As for the conflicts, just notice there won't be one, you're not editing the same files. I thought you were but both files just contain the same paragraph, that's my bad. 👍

Comment (safwansamsudeen) [reply building/tooling/analyzers/feedback-guidelines.md:8]: Ah, got it.

**Comment by safwansamsudeen** [comment]
Any update on this, folks?

File: building/github/contributors-pull-request-guide.md [hunk 12]
```diff
@@ -1,41 +1,40 @@
...
 
 Creating a pull request for a Concept Exercise or Practice Exercise can be daunting with the many rules around these types of exercises.
-For this reason, it can take a maintainer 2 to 3 hours to review and can result in dozens of comments.
-
+For this reason, it can take a maintainer two to three hours to review and can result in dozens of comments.  
 To help you, there'll be one primary reviewer commenting on your pull request.
 
 Don't be discouraged by the number of review comments; it is perfectly normal for an exercise to go through several rewrites before arriving at something both you and the primary reviewer are happy with.
 
 Here are some recommendations to help streamline the process even more.
```
Comment (ErikSchierboom) [inline building/github/contributors-pull-request-guide.md:18]: With the "Recommendation: " prefix removed, maybe we can add a recommendations header instead?

```suggestion
## Recommendations

Here are some recommendations to help streamline the process even more.
```

**Review by ErikSchierboom — COMMENTED** [review]
Just one last suggestion.

File: building/tooling/analyzers/feedback-guidelines.md (hunk 11 above)
Comment (iHiD) [reply building/tooling/analyzers/feedback-guidelines.md:8]: It should probably link here: `https://exercism.org/about/`

**Review by ErikSchierboom — APPROVED** [review]
